include/swift/SIL/SILFunction.h

```diff
@@ -430,6 +430,11 @@ class SILFunction
   /// method itself, the function can be referenced from vtables of derived
   /// classes in other compilation units.
   SILLinkage getEffectiveSymbolLinkage() const {
+    // These are implementations details that don't get special treatment: their
+    // stored linkage is the correct one.
+    if (isThunk() || isSpecialization())
```

Comment:
I think it would make more sense to turn this into an assertion and fix those places that create thunks no?

Reply:
Hm, I'm not sure: this is called in places that have just an arbitrary `SILFunction`:

https://github.com/apple/swift/blob/e8ee93b131d2ad360a8ff02034eb6c29718e411c/lib/IRGen/GenDecl.cpp#L1618-L1619

We could push this condition down to them, but I'm not sure it makes sense. To me, it seems reasonable to say that `getEffectiveSymbolLinkage` is the way to get the true IR/ELF/mach-O linkage, and thus having this split here is the right place.

Reply:
I'm suggesting that `getClassSubclassScope()` should not be set on thunks. Whoever is creating the thunk's SILFunction is also setting `getClassSubclassScope()`, which seems clearly wrong.

Reply:
Ah, I had misunderstood. That sounds good.